### PR TITLE
Prevent erasing _config that's been set in the constructor

### PR DESCRIPTION
**Pull request by Phillaf**
I made this change in PR #335 but it got dropped in issue ~~#368~~ #378 because I didnt write a test for it.

I added a test case for an inherited behavior. This doesn't break the test added for indexed config.

### CODE REVIEW

File: tests/TestCase/Model/Behavior/UploadBehaviorTest.php [hunk 1]
```diff
@@ -5,12 +5,22 @@
 use Cake\Event\Event;
 use Cake\ORM\Entity;
 use Cake\ORM\Table;
+use Cake\ORM\TableRegistry;
 use Cake\TestSuite\TestCase;
 use Josegonzalez\Upload\Model\Behavior\UploadBehavior;
 use ReflectionClass;
 
+class ChildBehavior extends UploadBehavior
```
Comment (josegonzalez) [inline tests/TestCase/Model/Behavior/UploadBehaviorTest.php]: Move this to a new file.

File: src/Model/Behavior/UploadBehavior.php [hunk 2]
```diff
@@ -34,7 +34,6 @@ public function initialize(array $config)
             }
         }
 
-        $this->_config = [];
```
Comment (josegonzalez) [inline src/Model/Behavior/UploadBehavior.php]: Do tests still pass if you don't remove this line?

Comment (Phillaf) [inline src/Model/Behavior/UploadBehavior.php]: The test that I just added fails when this line is present.

**Comment by Phillaf** [comment]
I made a typo in the original pull request description. The problematic line has been re-introduced in #378

File: tests/Fixture/FilesFixture.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @var array
+     */
+    public $records = [
+        ['filename' => 'FileOne'],
+        ['filename' => 'FileTwo'],
+        ['filename' => 'FileThree'],
+    ];
+
+    public function init()
```
Comment (josegonzalez) [inline tests/Fixture/FilesFixture.php]: Can you update test coverage so it ignores this file?

**Pull request merged**